feat: add `blas/base/cdotu`

gururaj1512 opened the pull request:
Resolves a part of  #2039.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds javascript implementation for `blas/base/cdotu`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Resolves a part of  #2039.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add blas/base/cdotu

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
add repl file

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

ShabiShett07 commented:
Hello Gururaj, I have left some comments in your PR #4633. Just so you know, the same will be applied here too. Let me know if any other error occurs

MeKaustubh07 committed:
bench: change variable names

MeKaustubh07 committed:
docs: improve JSDoc and Change variable naming

MeKaustubh07 committed:
Fixed repl and updated readme

MeKaustubh07 committed:
Updated test comments and renaming of variables

MeKaustubh07 committed:
add examples

MeKaustubh07 committed:
updated test and refactored bench

MeKaustubh07 committed:
chore: update documentation

MeKaustubh07 committed:
add Assign API

MeKaustubh07 committed:
refactor assign

MeKaustubh07 committed:
updated copyright years and use assign API

MeKaustubh07 committed:
use string format instead of concatination

MeKaustubh07 committed:
update description

MeKaustubh07 commented:
@kgryte Ready for Review

gururaj1512 commented on lib/node_modules/@stdlib/blas/base/cdotu/docs/repl.txt:
```diff
@@ -0,0 +1,108 @@
+    Computes the dot product of two single-precision complex floating-point
+    vectors using alternative indexing semantics.
+
+    While typed array views mandate a view offset based on the underlying
+    buffer, offset parameters support indexing based on starting indices.
+    Parameters
```
```suggestion
    buffer, offset parameters support indexing based on starting indices.

    Parameters
```

MeKaustubh07 committed:
commit

added space

MeKaustubh07 committed:
Merge remote-tracking branch 'upstream/develop' into blas-base-cdotu

MeKaustubh07 committed:
fix: correct description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
fix: add lint warning

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
fix: declare variable properly

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 committed:
fix: declare variable properly

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

MeKaustubh07 commented:
@kgryte ready for the  review

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/test/test.cdotu.js:
```diff
@@ -0,0 +1,217 @@
+	t.end();
+});
+
+tape( 'the function calculates the dot product of two single-precision complex floating-point strided arrays `x` and `y`', function test( t ) {
```
```suggestion
tape( 'the function calculates the dot product of two strided arrays `x` and `y`', function test( t ) {
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/test/test.js:
```diff
@@ -0,0 +1,82 @@
+		'@stdlib/utils/try-require': tryRequire
+	});
+
+	t.strictEqual( cdotu, main, 'returns expected value' );
```
```suggestion
	t.strictEqual( cdotu, main, 'returns JavaScript implementation' );
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/test/test.js:
```diff
@@ -0,0 +1,82 @@
+		'@stdlib/utils/try-require': tryRequire
+	});
+
+	t.strictEqual( cdotu, mock, 'returns expected value' );
```
```suggestion
	t.strictEqual( cdotu, mock, 'returns native implementation' );
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.js:
```diff
@@ -0,0 +1,110 @@
+
+var bench = require( '@stdlib/bench' );
+var uniform = require( '@stdlib/random/array/uniform' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.js:
```diff
@@ -0,0 +1,110 @@
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			d = cdotu( x.length, x, 1, y, 1 );
+			if ( isnan( real( d ) ) ) {
```
```suggestion
			if ( isnanf( real( d ) ) ) {
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.js:
```diff
@@ -0,0 +1,110 @@
+			}
+		}
+		b.toc();
+		if ( isnan( imag( d ) ) ) {
```
```suggestion
		if ( isnanf( imag( d ) ) ) {
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,110 @@
+
+var bench = require( '@stdlib/bench' );
+var uniform = require( '@stdlib/random/array/uniform' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,110 @@
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			d = cdotu( x.length, x, 1, 0, y, 1, 0 );
+			if ( isnan( real( d ) ) ) {
```
```suggestion
			if ( isnanf( real( d ) ) ) {
```

MeKaustubh07 commented on lib/node_modules/@stdlib/blas/base/cdotu/benchmark/benchmark.ndarray.js:
```diff
@@ -0,0 +1,110 @@
+			}
+		}
+		b.toc();
+		if ( isnan( imag( d ) ) ) {
```
```suggestion
		if ( isnanf( imag( d ) ) ) {
```

MeKaustubh07 committed:
Apply suggestions from code review

Co-authored-by: Kaustubh Patange <kaustubh.mp007@gmail.com>
Signed-off-by: Kaustubh Patange <kaustubh.mp007@gmail.com>
